src/appchain.cairo

```diff
@@ -144,29 +142,25 @@ pub mod appchain {
 
     #[abi(embed_v0)]
     impl Appchain of IAppchain<ContractState> {
-        fn update_state(
-            ref self: ContractState,
-            snos_output: Span<felt252>,
-            layout_bridge_output: Span<felt252>,
-        ) {
+        fn update_state(ref self: ContractState, layout_bridge_output: Span<felt252>) {
             self.reentrancy_guard.start();
             self.config.assert_only_owner_or_operator();
 
             let program_info = self.config.program_info.read();
 
+            let lb = deserialize_layout_bridge_output(layout_bridge_output);
+
             // StarknetOS (SNOS) proof is wrapped in bootloader so 3rd element is the program hash
             // of bootloaded program, in our case SNOS.
-            let snos_program_hash = snos_output.at(2);
             assert(
-                program_info.snos_program_hash == *snos_program_hash,
+                program_info.snos_program_hash == lb.bootloader_output.snos_program_hash,
                 errors::SNOS_INVALID_PROGRAM_HASH,
             );
 
             // Layout bridge program is also bootloaded, and the 3rd element is the hash of the
             // output of the program that has been bootloaded.
```

Comment:
This should be updated and put inside the deserialize_layout_bridge_output function, wdyt?